contracts/ronin/validator/CoinbaseExecution.sol

```diff
@@ -217,31 +218,40 @@ abstract contract CoinbaseExecution is
   function _distributeRewardToTreasuriesAndCalculateTotalDelegatingReward(
     uint256 lastPeriod,
     address[] memory currValidatorIds
-  ) private returns (uint256 totalDelegatingReward, uint256[] memory delegatingRewards) {
+  )
+    private
+    returns (uint256 totalDelegatingReward, uint256[] memory delegatingRewards, uint256[] memory delegatingFFRewards)
+  {
     address vId; // validator id
     address payable treasury;
-    delegatingRewards = new uint256[](currValidatorIds.length);
 
-    for (uint _i; _i < currValidatorIds.length;) {
-      vId = currValidatorIds[_i];
+    uint256 length = currValidatorIds.length;
```

Comment:
If it's trade off, cache the max commission rate here, instead of cache this length.

Reply:
stack too deep occur with `submitBlockReward` function